extensions/transferto/action.go

```diff
@@ -0,0 +1,141 @@
+package transferto
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/nyaruka/gocommon/urns"
+	"github.com/nyaruka/goflow/extensions/transferto/client"
+	"github.com/nyaruka/goflow/flows"
+	"github.com/nyaruka/goflow/flows/actions"
+	"github.com/nyaruka/goflow/flows/events"
+	"github.com/nyaruka/goflow/utils"
+
+	"github.com/shopspring/decimal"
+)
+
+func init() {
+	actions.RegisterType(TypeTransferAirtime, func() flows.Action { return &TransferAirtimeAction{} })
+}
+
+type transferToConfig struct {
+	APIToken string `json:"api_token"`
+	Account  string `json:"account"`
+	Currency string `json:"currency"`
+	Disabled bool   `json:"disabled"`
+}
+
+// TypeTransferAirtime is the type constant for our airtime action
+var TypeTransferAirtime = "transfer_airtime"
+
+// TransferAirtimeAction attempts to make a TransferTo airtime transfer to the contact
+type TransferAirtimeAction struct {
+	actions.BaseAction
+
+	Amounts map[string]decimal.Decimal `json:"amounts"`
```

Comment:
In legacy flows, we have a whole dict of information here about each country/currency - but most of that is for the UI, not the engine. So need to discuss where it belongs instead.

Reply:
We discussed and decided this should be currency->amount